### ISSUE TITLE
Add macOS support

### ISSUE THREAD

**Issue by Darkflame72** [revoltchat/desktop#16]
Support macOS by providing a pre-built app.

**Comment by Darkflame72** [comment]
I would be happy to build the app, you can also use GitHub actions to build it automatically on a release.

**Comment by StarWitch** [comment]
I'm also willing to help get a macOS release out. Do you have some basic build instructions available somewhere so I can save some time getting started?

**Comment by StarWitch** [comment]
Figured out the build steps.

Node v17 is broken and dies with `name 'openssl_fips' is not defined while evaluating condition 'openssl_fips != ""' in binding.gyp while trying to load binding.gyp`, but downgrading to Node v16 through Homebrew works. Doesn't seem like a Revolt problem so much as a Node problem based on [a StackOverflow post](https://stackoverflow.com/questions/69882740/how-to-rebuild-epoll-package-in-electron) I found.

Beyond that, this should work:

```bash
brew install node@16
export PATH="/usr/local/opt/node@16/bin:$PATH"
cd desktop/
npm install
export PATH=$PATH:$PWD/node_modules/.bin/
npm run build:bundle
electron-builder -m zip
```

This will spit out a `dist/Revolt-x.y.z-mac.zip` file. Unzip it and launch it (you might need to right-click->Open on the .app because macOS is a huge nag).

One final note is that you might run into issues with app signing if you haven't already set up Xcode with an independent developer signing key, but I don't really want to verify that. Just open up Xcode and make sure you have a signing key specified in there first. Shouldn't be strictly necessary but I wanted to mention that if anyone runs into problems.

**Comment by StarWitch** [comment]
Additional note: to build a *Universal* Mac `.app` for x86_64 and ARM64, you only need to change the last command:

```bash
electron-builder --mac --universal
```

**Comment by mau5atron** [comment]
@StarWitch thank you for linking the build steps! I made a build myself and released on my fork as a temporary placeholder until a Github Action is setup. I will run my Monterey VM later to build a universal release in case the one I provided has issues on newer hardware.

https://github.com/mau5atron/desktop/releases/tag/v1.0.2

I wanted to add that for anyone who wants to build themselves and doesn't want to run the code signing steps, they can set `CSC_IDENTITY_AUTO_DISCOVERY=false` to their environment and then run the build.
